### PR TITLE
Some improvements to JsonRepo and Repo

### PR DESCRIPTION
**Pull request by tejainece**

### CODE REVIEW

File: lib/src/json/json.dart [hunk 1]
```diff
@@ -3,38 +3,35 @@ library jaguar_serializer.json;
...
- * Same usage as [SerializerRepo]
- */
+/// Repository that serialize/deserialize JSON.
+///
+/// Same usage as [SerializerRepo]
 class JsonRepo extends SerializerRepo {
-  JsonRepo({String typeKey: defaultTypeInfoKey}) : super(typeKey: typeKey);
+  JsonRepo({List<Serializer> serializers, String typeKey: defaultTypeInfoKey})
+      : super(typeKey: typeKey) {
+    if (serializers is List) addAll(serializers);
```
Comment (lejard-h) [inline lib/src/json/json.dart]: we could do this in the constructor of `SerializerRepo` directly

File: lib/src/serializer/repo.dart [hunk 2]
```diff
@@ -137,42 +112,112 @@ class SerializerRepo {
...
 
-    if (serializer == null) {
-      throw new Exception("Cannot find serializer for type $type");
-    }
+    if (object is Map) {
+      final map = {};
+      object.forEach((key, value) {
+        final k = _to(key,
+            type: key.runtimeType, withType: withType, typeKey: typeKey);
+        if (value == null) {
```
Comment (lejard-h) [inline lib/src/serializer/repo.dart]: _to(value) with value == null should return null here no ?

Comment (lejard-h) [inline lib/src/serializer/repo.dart]: cleaner if `_to(value)` return null when value is null no ?

File: lib/src/serializer/repo.dart [hunk 3]
```diff
@@ -137,42 +112,112 @@ class SerializerRepo {
...
+        return _from2(decoded, typeKey, ser);
+      }
+    } else {
+      throw new Exception('Unknown type ${decoded.runtimeType}!');
     }
-    if (serializer == null) {
-      throw new Exception("Cannot find serializer for type $type");
+  }
+
+  dynamic _from2(Map decoded, String typeKey, Serializer ser) {
```
Comment (lejard-h) [inline lib/src/serializer/repo.dart]: can you choose more explicit name for `_from` and `_from2` ?

File: pubspec.yaml [hunk 4]
```diff
@@ -1,5 +1,5 @@
 name: jaguar_serializer
-version: 0.3.7
+version: 0.3.10
```
Comment (lejard-h) [inline pubspec.yaml]: 0.3.7 to 0.3.10 ?

Comment (tejainece) [inline pubspec.yaml]: There were intermediate version.

https://pub.dartlang.org/packages/jaguar_serializer#pub-pkg-tab-versions

File: lib/src/serializer/repo.dart (hunk 2 above)
Comment (tejainece) [inline lib/src/serializer/repo.dart]: My understanding is `_to(value)` returns null. Why i put the check there is you have `null.runtimeType` which returns `Null` probably not what we wany. Although, it produces expected results.

File: lib/src/json/json.dart (hunk 1 above)
Comment (tejainece) [inline lib/src/json/json.dart]: Fixed

**Comment by tejainece** [comment]
Fixed all issues

**Pull request merged**